src/libraries/System.Security.Cryptography/tests/AesGcmTests.cs

```diff
@@ -337,11 +360,18 @@ public static void AesGcmNistTests(AEADTest testCase)
             }
         }
 
-        [Theory]
+        [ConditionalTheory]
         [MemberData(nameof(GetNistGcmTestCases))]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/51332", TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.MacCatalyst)]
         public static void AesGcmNistTestsTamperTag(AEADTest testCase)
         {
+            if (PlatformDetection.IsOSX &&
+                testCase.Tag.Length != CryptoKitSupportedTagSizeInBytes &&
+                !PlatformDetection.OpenSslPresentOnSystem)
+            {
+                throw new SkipTestException("Platform does not support tag sizes other than 128-bit");
```

Comment:
Instead of skipping, these tests could check for the PNSE.

Reply:
The `ValidTagSize` does check the PNSE. I didn't feel it was worth checking PNSE in every test, but I don't mind doing it.